--- /content/drive/My Drive/efficientnet-pytorch/EfficientNet-PyTorch/EfficientNet-PyTorch-master/efficientnet_pytorch/utils.py
+++ /content/drive/My Drive/efficientnet-pytorch/EfficientNet-PyTorch/EfficientNet-PyTorch-master/efficientnet_pytorch/utils.py
@@ -1,5 +1,5 @@
 class Identity(nn.Module):
-    def __init__(self, ):
+    def __init__(self,):
         super(Identity, self).__init__()
 
     def forward(self, input):